### PR TITLE
8254265: s390 and linux 32 bit builds broken

### PR DESCRIPTION
**Pull request by TheRealMDoerr**
JDK-8253717 missed ocurrances of JavaThread::stack_guard_zone_size() which was moved to class StackOverflow.

I just verified s390 build. Can anybody check linux 32 bit?

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8254265](https://bugs.openjdk.java.net/browse/JDK-8254265): s390 and linux 32 bit builds broken

### Reviewers
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 6ec8881e21c94e84e7d117d007b69ef576f74ea4

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/568/head:pull/568`
`$ git checkout pull/568`

### CODE REVIEW

**Comment by coleenp** [comment]
I'm sorry, I really thought I'd built all the platforms.  Trying to figure out what happened.  This change looks good.  I'm trying to verify linux-32 bit now.

**Review by coleenp — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]

File: src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp [hunk 1]
```diff
@@ -860,7 +860,7 @@ void os::workaround_expand_exec_shield_cs_limit() {
    * we don't have much control or understanding of the address space, just let it slide.
    */
   char* hint = (char*)(Linux::initial_thread_stack_bottom() -
-                       (JavaThread::stack_guard_zone_size() + page_size));
+                       (StackOverflow::stack_guard_zone_size() + page_size));
```
Comment (coleenp) [inline src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp:863]: Lines 841 and 842 also need to be fixed.  Still testing.

**Review by coleenp — CHANGES_REQUESTED** [review]

**Comment by coleenp** [comment]
Also could you change this too?
diff --git a/src/hotspot/os/linux/os_linux.cpp b/src/hotspot/os/linux/os_linux.cpp
index 85560cc..cf20034 100644
--- a/src/hotspot/os/linux/os_linux.cpp
+++ b/src/hotspot/os/linux/os_linux.cpp
@@ -1935,7 +1935,7 @@ void * os::Linux::dll_load_in_vmthread(const char *filename, char *ebuf,
       StackOverflow* overflow_state = jt->stack_overflow_state();
       if (!overflow_state->stack_guard_zone_unused() &&     // Stack not yet fully initialized
           overflow_state->stack_guards_enabled()) {         // No pending stack overflow exceptions
-        if (!os::guard_memory((char *)jt->stack_end(), overflow_state->stack_guard_zone_size())) {
+        if (!os::guard_memory((char *)jt->stack_end(), StackOverflow::stack_guard_zone_size())) {
           warning("Attempt to reguard stack yellow zone failed.");
         }
       }

**Review by RealLucy — CHANGES_REQUESTED** [review]
Change looks good, but is complete only when you adapt the two lines mentioned by coleenp as well.
src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp:
      limit += JavaThread::stack_red_zone_size() +
        JavaThread::stack_yellow_zone_size();

**Comment by TheRealMDoerr** [comment]
Thanks for reviewing and checking linux 32 bit. x86_32 is not among the supported dev kit targets (https://openjdk.java.net/groups/build/doc/building.html) and I don't have a machine which is configured for that build, but maybe devkit does work.

**Review by coleenp — APPROVED** [review]
Thank you for fixing this!

**Comment by coleenp** [comment]
Actually the linux-x86 build is broken because of something in the libdwp native code right now but this code compiles.
https://bugs.openjdk.java.net/browse/JDK-8254270

**Pull request closed without merging**